### PR TITLE
Update some parallel documentation

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Some of it was out of date. Also added some info and warnings.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: docs/DevGuide/Parallelization.md [hunk 1]
```diff
@@ -111,14 +111,30 @@ phase is determined and the `execute_next_phase` function is called on
...
+implies some error in the order messages have been sent/received. For example,
+if core 0 was paused and waiting to receive a message from core 1, but core 1
+was also paused and waiting to receive a message from core 0, this would be
+considered a deadlock. We detect deadlocks during the `Exit` phase. All
+executables have the option to specify a function with the following signature
+
+\snippet Test_DetectHangArray.cpp deadlock_analysis_function
+
+If this function is specified in the metavariables and a deadlock occurs, this
+function and all the simple actions in it will run. The information printed
```
Comment (wthrowe) [inline docs/DevGuide/Parallelization.md:132]: What do you mean by "simple actions in [a function]"?

File: docs/DevGuide/Parallelization.md [hunk 2]
```diff
@@ -806,15 +806,15 @@ will be invoked; in this case the callback function re-runs
 \snippet Test_AlgorithmGlobalCache.cpp check_mutable_cache_item_is_ready
 
 Note that `Parallel::mutable_cache_item_is_ready` is called on a local
-core and does no parallel communication.
+node and does no parallel communication.
```
Comment (wthrowe) [inline docs/DevGuide/Parallelization.md:809]: "_the_ local node".  (There's only one.)

File: docs/DevGuide/Parallelization.md [hunk 3]
```diff
@@ -806,15 +806,15 @@ will be invoked; in this case the callback function re-runs
...
 
 ### 2. Retrieving the item
 
 The item is retrieved using `Parallel::get` just like for constant items.
 For example, to retrieve the item `Tags::VectorOfDoubles`:
 \snippet Test_AlgorithmGlobalCache.cpp retrieve_mutable_cache_item
 
-Note that `Parallel::get` is called on a local core and does no
+Note that `Parallel::get` is called on a local node and does no
```
Comment (wthrowe) [inline docs/DevGuide/Parallelization.md]: "the" again.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: docs/DevGuide/Parallelization.md (hunk 1 above)
Comment (knelli2) [reply docs/DevGuide/Parallelization.md:132]: The function named `run_deadlock_analysis_simple_actions` calls simple actions.

Comment (wthrowe) [reply docs/DevGuide/Parallelization.md:132]: OK, yeah, I guess that should have been clear if I'd paid more attention.

**Comment by knelli2** [comment]
Squashed the "the"s.

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**